src/pfe/file-watcher/server/src/projects/ShellExtensionProject.ts

```diff
@@ -208,26 +234,35 @@ export class ShellExtensionProject implements IExtensionProject {
 
     /**
      * @function
-     * @description Get the build log for the project.
+     * @description Get logs from files or directories.
      *
-     * @param logDirectory <Required | string> - The log location directory.
+     * @param type <Required | String> - The type of log ("build" or "app")
+     * @param logDirectory <Required | String> - The log location directory.
+     * @param projectID <Required | String> - An alphanumeric identifier for a project.
+     * @param containerName <Required | String> - The docker container name.
      *
-     * @returns Promise<BuildLog>
+     * @returns Promise<Array<AppLog | BuildLog>>
      */
-    getBuildLog = async (logDirectory: string): Promise<BuildLog> => {
-        return await logHelper.getBuildLogs(logDirectory, this.config.buildLogs);
+    getLogs = async (type: string, logDirectory: string, projectID: string, containerName: string): Promise<Array<AppLog | BuildLog>> => {
+        if (type.toLowerCase() != "build" && type.toLowerCase() != "app") return;
+        this.setLogsOriginFromExtension();
+        return await logHelper.getLogs(type, logsOrigin, logDirectory, projectID, containerName);
     }
 
-    /**
-     * @function
-     * @description Get the app log for the project.
-     *
-     * @param logDirectory <Required | String> - The log location directory.
-     *
-     * @returns Promise<AppLog>
-     */
-    getAppLog = async (logDirectory: string): Promise<AppLog> => {
-        return await logHelper.getAppLogs(logDirectory, this.config.appLogs);
+    setLogsOriginFromExtension(): void {
+        if (this.config.buildContainerLogs) {
```

Comment:
Should we do some sanitization or error check on these properties here, to guard against extensions that define these incorrectly?  (if you're doing that in `logHelper.getLogs` then it's fine not to do it here)

Reply:
yup, its all typed against the interfaces in `logHelper`, so if the type check fails it will fail

Reply:
But Typing in `ts` is only for compile time checking.  This doesn't guard at runtime.